pennylane/estimator/compact_hamiltonian.py

```diff
@@ -34,6 +35,26 @@ class CDFHamiltonian:
 
     num_orbitals: int
     num_fragments: int
+    one_norm: float | None = None
+
+    def __post_init__(self):
+        """Checks the types of the inputs."""
+
+        if not isinstance(self.num_orbitals, int):
+            raise TypeError(
+                f"num_orbitals must be an integer, got {type(self.num_orbitals).__name__}"
+            )
+        if not isinstance(self.num_fragments, int):
+            raise TypeError(
+                f"num_fragments must be an integer, got {type(self.num_fragments).__name__}"
+            )
+        if self.one_norm is not None and not isinstance(self.one_norm, (float, int)):
+            raise TypeError(
+                f"one_norm must be a float or integer (or None), "
+                f"but received type {type(self.one_norm).__name__}"
+            )
```

Comment:
If we are going to do input validation, then we should also check that it's greater than or equal to 0. 
```suggestion
        if self.one_norm is not None and not (isinstance(self.one_norm, (float, int)) and self.one_norm >= 0):
            raise TypeError(
                f"one_norm must be a positive float or integer (or None), "
                f"but received type {type(self.one_norm).__name__}"
            )
```